app/src/main/java/org/thoughtcrime/securesms/ui/color/Colors.kt

```diff
@@ -250,12 +187,39 @@ fun Colors.radioButtonColors() = RadioButtonDefaults.colors(
     disabledColor = disabled
 )
 
-fun TextSecurePreferences.Companion.getAccentColor(context: Context): Color = when (getAccentColorName(context)) {
-    BLUE_ACCENT -> primaryBlue
-    PURPLE_ACCENT -> primaryPurple
-    PINK_ACCENT -> primaryPink
-    RED_ACCENT -> primaryRed
-    ORANGE_ACCENT -> primaryOrange
-    YELLOW_ACCENT -> primaryYellow
-    else -> primaryGreen
+@Composable
+fun transparentButtonColors() = ButtonDefaults.buttonColors(backgroundColor = Color.Transparent)
+
+@Composable
+fun destructiveButtonColors() = ButtonDefaults.buttonColors(backgroundColor = Color.Transparent, contentColor = LocalColors.current.danger)
+
+
+/**
+ * This class holds two instances of [Colors], [light] representing the [Colors] to use when the system is in a
+ * light theme, and [dark] representing the [Colors] to use when the system is in a dark theme.
+ *
+ * If the user has [followSystemSettings] turned on then [light] should be equal to [dark].
+ */
+data class LightDarkColors(
+    val light: Colors,
+    val dark: Colors
+) {
+    @Composable
+    fun colors() = if (light == dark || isSystemInDarkTheme()) dark else light
+}
+
+/**
+ * Courtesy constructor that sets [light] and [dark] based on properties.
+ */
+fun LightDarkColors(isClassic: Boolean, isLight: Boolean, followSystemSettings: Boolean, primaryOrUnspecified: Color): LightDarkColors {
```

Comment:
I did a lot of clean up in my PR, which is now tricky to add in with all the new directions.
the logic I added was there to decouple all the classic references to future proof the themes.

Reply:
We don't really need to future-proof. We have 4 times as many themes as most apps, and I doubt it'd be a priority to add more.

Anyway, this is a courtesy function to create `LightDarkColors`, not deep inside business logic.

It's much the same logic as how you had `oceanTheme = ColorSet(oceanLight, oceanDark)`.

I was thinking we would drop your color commit, and you can open a PR for the typography commit after this. At the moment we only have one set of typography, and one set of shapes, so it's not super pressing to have them be switchable, when there's nothing to switch to as-of-yet.

Reply:
Future proofing is important especially at such a low cost ( or with the changes already made in this case ).
I'm happy to recreate the changes in a separate branch later if this is the issue, though I thought it was easy enough to integrate here.
I can do it with a Typography PR later.

Reply:
It's not really tightly coupled, this is just a courtesy function.

Reply:
Do you want the `component library` logic and class defs to be in a different package to the app specific `Colors` implementations etc?

Because I'm not really sure what is so different.

Neither of our code could be used in another app as-is.

Neither could immediately take on a 3rd theme without a little bit of work.

But I don't think that matters.

Reply:
All good, let's focus on onboarding and we'll do some theming work separately.